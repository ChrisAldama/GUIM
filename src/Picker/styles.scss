@import '../styles/colors.scss';
@import '../styles/themes.scss';

.GUIMPicker {
  display: flex;
  align-items: center;
  text-align: center;
  cursor: pointer;
  height: 30px;
  & a {
    border: 1px solid var(--main-gray);
    color: var(--dark-gray);
    display: flex;
    background-color: var(--white);
    width: calc(51% - 1px);
    justify-content: center;
    align-items: center;
    transition: all 0.3s ease;
    height: 100%;
    &:last-child{
      border-top-right-radius: 3px;
      border-bottom-right-radius: 3px;
    }
    &:first-child{
      border-top-left-radius: 3px;
      border-bottom-left-radius: 3px;
    }
  }
}

.GUIMPickerBlue {
  & a:hover {
    @apply --blue-theme-hover;
  }
  & .active {
    @apply --blue-theme;
  }
}
.GUIMPickerGray {
  & a:hover {
    @apply --gray-theme-hover;
  }
  & .active {
    @apply --gray-theme;
  }
}
.GUIMPickerGreen {
  & a:hover {
    @apply --green-theme-hover;
  }
  & .active {
    @apply --green-theme;
  }
}
