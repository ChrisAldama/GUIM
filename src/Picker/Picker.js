import React from 'react';
import './styles.scss';

const Pill = props => (
  <a
    onClick={props.select}
    className={props.active ? 'active' : ''}
    key={props.value}
  >
    {props.label}
  </a>
);

const Picker = props => (
  <div
    name={props.name}
    className={`GUIMPicker ${props.className} ${themes[props.color] || themes.blue}`}
  >
    {
      props.options.map(e =>
        (<Pill
          key={e.value}
          select={props.onChange.bind(null, e.label, e.value)}
          active={props.active.indexOf(e.value) !== -1}
          value={e.value}
          label={e.label}
        />))
    }
  </div>
);

Picker.defaultProps = {
  name: 'picker',
  onChange: (label, value) => console.log(`Picker:active => label: ${label}, value: ${value}`),
  options: [
    { label: 'Uno', value: 1 },
    { label: 'Dos', value: 2 },
  ],
  className: '',
  color: 'blue',
  guimInput: 'picker',
  active: [1],
};

const themes = {
  blue: 'GUIMPickerBlue',
  gray: 'GUIMPickerGray',
  green: 'GUIMPickerGreen',
};

export default Picker;
