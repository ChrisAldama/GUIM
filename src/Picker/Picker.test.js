import React from 'react';
import Picker from './Picker';
import renderer from 'react-test-renderer';
import { mount, shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('<Picker />', () => {
  describe('behaviour tests', () => {
    let active = 1;
    let component;
    let tree;
    let reRender;
    const makeRenderer = component =>
      () => tree = component.toJSON();

    beforeEach(() => {
      component = renderer.create(
        <Picker
          onChange={(label, value) => {active = value}}
          options={[
            {label: 'Uno', value: 1},
            {label: 'Dos', value: 2},
            {label: 'Tres', value: 3},
          ]}
          active={active}
        />);
      reRender = makeRenderer(component);
      reRender();
    })

    it('match snapshot', () => {
      expect(tree).toMatchSnapshot();
    });


    it('selects the right element', () => {
      tree.children[1].props.onClick();
      reRender();
      expect(active).toBe(2);

      tree.children[2].props.onClick();
      reRender();
      expect(active).toBe(3);
    });
  });

  describe('dom tests', () => {
    let active = 1;
    let component;
    let update = () => {
      component = mount(
        <Picker
          onChange={(label, value) => {active = value}}
          options={[
            {label: 'Uno', value: 1},
            {label: 'Dos', value: 2},
            {label: 'Tres', value: 3},
          ]}
          active={active}
        />);
    }
    beforeEach(() => {
      update();
    });

    it('changes the active className to the selected pill', () => {
      component.find('a').at(2).simulate('click');
      update();
      expect(component.find('.active').key()).toBe('3');
    });
  });
});
