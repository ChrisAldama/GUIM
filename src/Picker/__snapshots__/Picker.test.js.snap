// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Picker /> tests for multiple behaviour match snapshot 1`] = `
<div
  className="GUIMPicker  GUIMPickerBlue"
  name="picker"
>
  <a
    className="active"
    onClick={[Function]}
  >
    Uno
  </a>
  <a
    className=""
    onClick={[Function]}
  >
    Dos
  </a>
  <a
    className=""
    onClick={[Function]}
  >
    Tres
  </a>
</div>
`;

exports[`<Picker /> tests for normal behaviour match snapshot 1`] = `
<div
  className="GUIMPicker  GUIMPickerBlue"
  name="picker"
>
  <a
    className="active"
    onClick={[Function]}
  >
    Uno
  </a>
  <a
    className=""
    onClick={[Function]}
  >
    Dos
  </a>
  <a
    className=""
    onClick={[Function]}
  >
    Tres
  </a>
</div>
`;
